src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/PublisherAddress.cs

```diff
@@ -59,10 +58,74 @@ private PublisherAddress()
             {
                 return instances.Select(addressResolver);
             }
-            var result = await instanceResolver(endpoint).ConfigureAwait(false);
+            var result = instanceResolver(endpoint);
             return result.Select(addressResolver);
         }
 
+        /// <summary>Returns a string that represents the current object.</summary>
+        /// <returns>A string that represents the current object.</returns>
+        public override string ToString()
+        {
+            if (endpoint != null)
+            {
+                return endpoint;
+            }
+            if (instances != null)
+            {
+                return string.Join(", ", instances.Select(x => $"[{x.ToString()}]").OrderBy(x => x));
+            }
+            return string.Join(", ", addresses.Select(x => $"<{x}>").OrderBy(x => x));
+        }
+
+        bool Equals(PublisherAddress other)
+        {
+            return CollectionEquals(addresses, other.addresses) 
```

Comment:
I wonder whether we actually need the collections since we support registering multiple publishers? (probably out of scope of this PR)